js/plugin/post-edit.js

```diff
@@ -1,170 +1,144 @@
-			app.inputTemplateEl.setAttribute('value', '');
-			app.inputChartUrlEl.setAttribute('value', '');
-			app.inputChartIdEl.setAttribute('value', '');
-		},
-
-		openModal : function(){
-			if ( ! app.modalInitialized ) {
-				app.modalElements.container = app.modalElements.container.replace('{{iframeSrc}}', WPSimplechartBootstrap.appUrl);
-				app.modalElements.container = app.modalElements.container.replace('{{closeModal}}', WPSimplechartBootstrap.closeModal);
-				$( 'body' ).append( app.modalElements.container + app.modalElements.backdrop );
-				$( '#simplechart-close' ).click( function( e ) {
-					e.preventDefault();
-					if ( app.savedChart || confirm( app.confirmNoDataMessage ) ) {
-						$( '#simplechart-backdrop, #simplechart-modal' ).hide();
-						app.savedChart = false;
-					}
-				} );
-
-				app.modalInitialized = true;
-			} else {
-				$( '#simplechart-backdrop, #simplechart-modal' ).show();
+			if ( savedChart || confirm( confirmNoDataMessage ) ) {
+				hideModal();
+				savedChart = false;
 			}
-		},
-
-		/*
-		 * postMessage send/receive functions
-		 */
-		receiveMessages: function( e ) {
-			if ( _.isUndefined( e.data.src ) || 'simplechart' !== e.data.src ) {
-				return;
-			}
-
-			app.appOrigin = app.appOrigin || app.setAppOrigin();
-			if ( e.origin !== app.appOrigin ){
-				throw( 'Illegal Simplechart postMessage from ' + e.appOrigin );
-				return;
-			}
-
-			if ( app.isFrameReadyMessage( e.data ) ) {
-				console.log( 'window received ready message from Simplechart iframe');
-				app.childWindow = app.childWindow || document.getElementById('simplechart-frame').contentWindow;
-				app.sendSimplechartOptions();
-				app.sendSavedData();
-				return;
-			}
-
-			// parse data for iframe
-			app.chartData = JSON.parse( e.data.data.chartData );
-
-			// store template string
-			app.inputTemplateEl.value = app.chartData.template;
-
-			// store rest of JSON for chart
-			delete app.chartData.template;
-			app.inputEl.value = JSON.stringify( app.chartData );
+		} );
+		$( '#simplechart-launch' ).click( openModal );
+	}
 
-			// store base64 image string
-			app.imgInputEl.value = e.data.data.chartImg;
+	/**
+	 * Reopens already-rendered modal
+	 */
+	function openModal() {
+		$( '#simplechart-backdrop, #simplechart-modal' ).show();
+	}
 
-			// store published chart URL
-			if ( ! _.isUndefined( app.chartData.chartUrl ) ) {
-				app.inputChartUrlEl.value = app.chartData.chartUrl;
-			}
+	/**
+	 * Hide modal
+	 */
+	function hideModal() {
+		$( '#simplechart-backdrop, #simplechart-modal' ).hide();
+	}
 
-			// store published chart ID
-			if ( ! _.isUndefined( app.chartData.id ) ) {
-				app.inputChartIdEl.value = app.chartData.id;
-			}
+	/**
+	 * Extract messageType string when a postMessage is received
+	 */
+	function getMessageType(evt) {
+		// confirm same-origin or http(s)://localhost:8080
+		if ( evt.origin !== window.location.origin && !/https?:\/\/localhost:8080/.test(evt.origin )) {
```

Comment:
inconsistent spacing inside parentheses for this conditional and in the call to the `test()` method. I believe our JS practice is no spaces, but either way it should be consistent.